class-6/js-function.js

```diff
@@ -0,0 +1,77 @@
+//What is Pure function?
+//A pure function is a function that has the following properties:
+//It returns the same result if given the same arguments
+//It does not cause any observable side effects
+//It does not read external files
+//It does not rely on a random number generator
+
+//What is Higher Order Function?
+//A higher order function is a function that takes a function as an argument, or returns a function.
+
+
+//What is Call, Blind and Apply?
+//Call, Blind and Apply are the methods of Function.prototype.
+//Call and Apply are very similar in their usage except a little difference.
+//Call takes the arguments separately.
+//Apply takes the arguments as an array.
+//Bind returns a new function, allowing you to pass any number of arguments.
+
+//-----------------------------PURE FUNCTION---------------------------------
+function finalResult(quiz, assignment, finalExam) {
+    let total = quiz + assignment + finalExam;
+    return total;
+}
+console.log("You have got " + finalResult(10, 20, 30));
+// Output:
+// [Running] node "c:\Users\ahmed\Downloads\404RequestedUserNotFound\learnathon-2-javascript\class-6\js-function.js"
+// You have got 60
+//-----------------------------PURE FUNCTION END---------------------------------
+
+//-----------------------------HIGHER ORDER FUNCTION---------------------------------
```

Comment:
This is not a Higher order function example. Would you please check this link: https://www.freecodecamp.org/news/higher-order-functions-in-javascript-explained/